File: ValidParentheses/validParentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
  
  let i = 0;
  let stack = [];
  let answer = true;
  while (i < s.length) {
    if (open.includes(s[i])) {
      stack.push(open.indexOf(s[i]));
    }

    if (close.includes(s[i])) {
      if (close.indexOf(s[i]) == stack[stack.length - 1]) {
        stack.pop();
      } else {
        answer = false;
        break;
      }
    }
    i++;
  }

  if (stack.length != 0) {
    answer = false;
  }
  
  return answer
};

const open = ["(","[","{"];
const close = [")", "]", "}"];


const s = "["

console.log(isValid(s));

File: ValidParentheses/validParentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        openDict = {"(": 0, "[": 1, "{": 2}
        closeDict = {")": 0, "]": 1, "}": 2}
        openArr = []
        for i in s:
            if i in openDict:
                openArr.append(openDict[i])
                print("open", openArr)
            if i in closeDict:
                if len(openArr) == 0:
                    return False
                if (openArr[-1] != closeDict[i]):
                    return False
                openArr.pop()
        if len(openArr) > 0:
            return False
        return True
    

s = "(])"
sol = Solution()
print(sol.isValid(s))